### PR TITLE
Add pre-commit

### PR DESCRIPTION
**Pull request by scasagrande**
`pylint` is annoying

Manual formatting is annoying

This PR replaces `pylint` with `black`, and manages the execution of this tool with the wonderful `pre-commit`. I've had a lot of success with this in other projects so I'm starting the migration here as well. For the first PR, we're adding `pre-commit` and `black`, running the tool against the entire repo, updating CI to run these checks, and updating a few lines in the docs.

Also, it looks like `pysim-visa` isn't well defined. The latest version, `0.4.0`, has some dependencies on `six` but that's not included in the package definition. It looks like that was fixed afterwards, but isn't in a cut release. This wasn't a problem before because `pylint` was bringing in `six`. For now I've added it as a dev dependency, but it'll have to be correctly fixed in the future.

### CODE REVIEW

File: doc/source/devguide/index.rst [hunk 1]
```diff
@@ -22,8 +22,19 @@ Getting Started
...
+This repo also contains a series of static code checks that are managed
+via `pre-commit`. This tool, once setup, will manage running all of these
+checks prior to each commit on your local machine.::
+
+$ pip install pre-commit
+$ pre-commit install
+
+These checks are also run in CI, and must pass in order to generate
+a passing build. It is suggested that you install the git hooks, but
+they can be run manually on all files. See the `pre-commit` homepage
```
Comment (trappitsch) [inline doc/source/devguide/index.rst:36]: Would be nice to put a link here for the user: https://pre-commit.com/#3-install-the-git-hook-scripts
The `pre-commit install` does the installation, correct?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    rev: v4.1.0
+    hooks:
+    - id: no-commit-to-branch
+      args: [--branch, master]
+    - id: trailing-whitespace
+    - id: end-of-file-fixer
+    - id: check-yaml
+    - id: debug-statements
+  - repo: https://github.com/psf/black
+    rev: 21.12b0
```
Comment (trappitsch) [inline .pre-commit-config.yaml:12]: Might be useful to put the pinned black version into the docs as well somewhere or mention it...

**Review by trappitsch — COMMENTED** [review]
Love it! Agree with all your statements to go for it :)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (scasagrande) [reply .pre-commit-config.yaml:12]: Ehhh, it'll be updated periodically. It should be sufficiently self-documented. I briefly put in the docs that you should be using pre-commit for formatting, and it'll automatically update the plugins when they change.

I'll put something in the readme though, that's a good idea

File: doc/source/devguide/index.rst (hunk 1 above)
Comment (scasagrande) [reply doc/source/devguide/index.rst:36]: Yes, `pre-commit install` installs the hooks. The cool thing is that as the configuration file changes and updates, the tool automatically updates all the plugins!

**Comment by scasagrande** [comment]
:ship: :ship: :ship:

**Pull request merged**